From 629e05d8dc795a70fd2bcd3d0301641105bf0b06 Mon Sep 17 00:00:00 2001
From: John Dennis <jdennis@redhat.com>
Date: Wed, 15 Jun 2016 11:50:24 -0400
Subject: [PATCH] enable user supplied CFLAGS
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

CFLAGS is initialized to the empty string in configure.ac, this
effectively turned off user supplied values for CFLAGS preventing site
specific values from being used. A further complicating factor was of
all the user supplied values documented in Automake only CFLAGS was
disabled allowing all other user supplied variables to take
effect. Some variables must be coordinated (e.g. CFLAGS with LDFLAGS),
the fact LDFLAGS was picked up from the environment but CFLAGS was
discarded caused build failures due to incompatible combination of
compiler and linker options.

The problem was first introduced in commit: 73d9c98f "Reset CFLAGS
when --enable-debugging is used". This patch simply removes hardcoding
CFLAGS to the empty string and appends the debug options
(--enable-debugging) to the existing CFLAGS.

Proper use of the variables is described in the Automake documentation
in the section "Flag Variables Ordering"
https://www.gnu.org/software/automake/manual/html_node/Flag-Variables-Ordering.html

Although the Automake documentation claims manipulating CFLAGS
directly is improper use there are many examples of this in the
existing configure.ac, this patch makes no attempt at addressing this
issue, rather it makes existing usage consistent. In the particular
case of debug flags appending to CFLAGS is probably the only valid
solution because the debug flags must appear at the end of the list of
flags in order to override earlier flags, CFLAGS always appears last
in the Makefile (see above Automake doc).

Signed-off-by: John Dennis <jdennis@redhat.com>
License: MIT
---
 configure.ac | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7c58870..cf86262 100644
--- a/configure.ac
+++ b/configure.ac
@@ -91,7 +91,6 @@ dnl
 dnl Check for programs
 dnl
 AC_PROG_CC
-CFLAGS=""
 AM_CFLAGS=""
 AC_HEADER_STDC
 LT_AC_PROG_RC
@@ -702,7 +701,7 @@ AC_ARG_ENABLE(debugging,   [  --enable-debugging      enable debuging optimizati
 if test "z$enable_debugging" = "zyes" ; then
     enable_debugging=yes
     LASSO_DEFINES="$LASSO_DEFINES -DLASSO_DEBUG"
-    AM_CFLAGS="-O0 -g -Wall -Wextra -Werror"
+    CFLAGS="$CFLAGS -O0 -g -Wall -Wextra -Werror"
 else
     enable_debugging=no
 fi    
-- 
2.5.5

